src/core/models/inputs/utxo_input.h

```diff
@@ -26,6 +27,7 @@ typedef struct {
   byte_t tx_id[IOTA_TRANSACTION_ID_BYTES];  ///< The BLAKE2b-256 hash of the transaction payload containing the
                                             ///< referenced output.
   uint16_t output_index;                    ///< The output index of the referenced output.
+  utxo_output_t *output;                    ///< Unspent output object. It is needed for calculating inputs commitment.
```

Comment:
```suggestion

```

basically, the input list is selected outputs which we can just use `utxo_outputs_list_t` as input in tx essence.

```
typedef struct {
  ...
  utxo_outputs_list_t* inputs;
  byte_t inputs_commitment[CRYPTO_BLAKE2B_HASH_BYTES];
  utxo_outputs_list_t* outputs;
  ...
} transaction_essence_t;
```

Reply:
never mind, output response is not the case. :P